src/Temporalio/Client/TemporalClient.Workflow.cs

```diff
@@ -220,6 +221,128 @@ public override async Task<TResult> QueryWorkflowAsync<TResult>(QueryWorkflowInp
                     resp.QueryResult.Payloads_).ConfigureAwait(false);
             }
 
+            /// <inheritdoc />
+            public async override Task<WorkflowUpdateHandle<TResult>> StartWorkflowUpdateAsync<TResult>(
+                StartWorkflowUpdateInput input)
+            {
+                // Build request
+                var req = new UpdateWorkflowExecutionRequest()
+                {
+                    Namespace = Client.Options.Namespace,
+                    WorkflowExecution = new()
+                    {
+                        WorkflowId = input.Id,
+                        RunId = input.RunId ?? string.Empty,
+                    },
+                    Request = new()
+                    {
+                        Meta = new()
+                        {
+                            UpdateId = input.Options?.UpdateID ?? Guid.NewGuid().ToString(),
+                            Identity = Client.Connection.Options.Identity,
+                        },
+                        Input = new() { Name = input.Update },
+                    },
+                };
+                if (input.Options is { } options)
+                {
+                    if (options.FirstExecutionRunId is { } firstExecutionRunId)
+                    {
+                        req.FirstExecutionRunId = firstExecutionRunId;
+                    }
+                    if (options.WaitForStage != UpdateWorkflowExecutionLifecycleStage.Unspecified)
+                    {
+                        req.WaitPolicy = new() { LifecycleStage = options.WaitForStage };
+                    }
+                }
+                if (input.Args.Count > 0)
+                {
+                    req.Request.Input.Args = new Payloads();
+                    req.Request.Input.Args.Payloads_.AddRange(
+                        await Client.Options.DataConverter.ToPayloadsAsync(
+                            input.Args).ConfigureAwait(false));
+                }
+                if (input.Headers != null)
+                {
+                    req.Request.Input.Header = new();
+                    req.Request.Input.Header.Fields.Add(input.Headers);
+                    // If there is a payload codec, use it to encode the headers
+                    if (Client.Options.DataConverter.PayloadCodec is IPayloadCodec codec)
+                    {
+                        foreach (var kvp in req.Request.Input.Header.Fields)
+                        {
+                            req.Request.Input.Header.Fields[kvp.Key] =
+                                await codec.EncodeSingleAsync(kvp.Value).ConfigureAwait(false);
+                        }
+                    }
+                }
+                // Invoke
+                var resp = await Client.Connection.WorkflowService.UpdateWorkflowExecutionAsync(
+                    req, DefaultRetryOptions(input.Options?.Rpc)).ConfigureAwait(false);
```

Comment:
> but eventually we should automatically start polling IMHO

Disagree. We decided as a team that start will default to `Accepted`. Therefore this call should error if your stage is not reached in a single API call. Granted this error can contain the handle in it for the _user_ to start polling (called via get result). The start call should never poll I don't think, that's what the get-result call does. If you choose to set `WaitForStage` as `Completed` you are saying "error if it doesn't complete within the single call". Otherwise you can use an alternative.

Reply:
> We decided as a team that start will default to Accepted.

Yes, considering that `Admitted` isn't yet available, but I can see arguments both ways.

> Therefore this call should error (or return unset handle?) if your stage is not reached in a single API call.

I would agree if the API didn't have a wait condition, it's not just a start call, it's start and wait.

I think we shouldn't expose `StartUpdate` with `WaitForStage=Completed`, you might as well use `ExecuteUpdate` and give us a timeout and we'll poll if necessary and the timeout permits.

For start with `Accepted`, I think we should poll if the user is willing to wait, why would the user care how many RPCs we make internally, what they care about is that it happens within a given timeout.
When using `Admitted`, there's obviously no need to poll.

> When polling is stable server side we should discuss what we want behavior here to be.

We can defer this conversation for when polling is stable though.

Reply:
> I think we shouldn't expose StartUpdate with WaitForStage=Completed, you might as well use ExecuteUpdate and give us a timeout and we'll poll if necessary and the timeout permits.

You don't think there are any valid use cases for wanting to get a handle back in that situation rather than a result? I know they have the option of specifying an updateID up front and using it to obtain a handle, but I'm not entirely convinced that it's unreasonable to allow the handle to be obtained straightforwardly.

> why would the user care how many RPCs we make internally

It's not implausible that high-volume users would want control over egress costs.

Reply:
Ok, there is a case for no enum today until admitted is implemented, and no completed ever (that's the difference between start and execute). Discussing with team.

Reply:
Removed publicly visible enum (except for interceptor which needs it still). Will defer polling until it's stable.